### PR TITLE
chore: Secret Validation change

### PR DESCRIPTION
**Pull request by sfc-gh-fbudzynski**
## Changes
- changed secret validation to use generated validation without manual adjustments
- changed parsing StringArrays to use `ParseCommaSeparatedStringArray()`
- removed known issue with ConflictingFields from sdk generator README
- removed comments and replaced them with jira issues

### CODE REVIEW

File: pkg/sdk/testint/secrets_gen_integration_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func TestInt_Secrets(t *testing.T) {
 
 	integrationId := testClientHelper().Ids.RandomAccountObjectIdentifier()
 
-	// "YYYY-MM-DD" or "YYYY-MM-DD HH-MI-SS" format has to be used, otherwise Snowflake returns error: "Invalid date/time format"
+	// "YYYY-MM-DD" or "YYYY-MM-DD HH:MI-SS" format has to be used, otherwise Snowflake returns error: "Invalid date/time format"
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/testint/secrets_gen_integration_test.go:22]: Nit: just asking: shouldn't it be HH:MI:SS?

File: pkg/sdk/testint/secrets_gen_integration_test.go [hunk 2]
```diff
@@ -121,7 +120,7 @@ func TestInt_Secrets(t *testing.T) {
 		})
 	})
 
-	// regarding the https://docs.snowflake.com/en/sql-reference/sql/create-secret secret with empty oauth_scopes list should inherit scopes from security_integration, but it does not
+	// TODO [SNOW-1678756]: oauth_copes not inherited or not displayed correctly when not provided
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/testint/secrets_gen_integration_test.go:123]: Nit: oauth_scopes

File: pkg/sdk/secrets_gen_test.go [hunk 3]
```diff
@@ -196,26 +196,46 @@ func TestSecrets_Alter(t *testing.T) {
...
-		opts.Set.SetForOAuthAuthorizationFlow = &SetForOAuthAuthorizationFlow{OauthRefreshToken: String("foo"), OauthRefreshTokenExpiryTime: String("bar")}
-		opts.Set.SetForBasicAuthentication = &SetForBasicAuthentication{Username: String("foo"), Password: String("bar")}
-		opts.Set.SetForGenericString = &SetForGenericString{SecretString: String("secret")}
-		assertOptsInvalidJoinedErrors(t, opts, errOneOf("AlterSecretOptions.Set", "SetForOAuthClientCredentialsFlow", "SetForOAuthAuthorizationFlow", "SetForBasicAuthentication", "SetForGenericString"))
+		opts.Set.SetForFlow = &SetForFlow{}
+		opts.Set.SetForFlow.SetForOAuthClientCredentials = &SetForOAuthClientCredentials{OauthScopes: &OauthScopesList{[]ApiIntegrationScope{{Scope: "foo"}}}}
+		opts.Set.SetForFlow.SetForOAuthAuthorization = &SetForOAuthAuthorization{OauthRefreshToken: String("foo"), OauthRefreshTokenExpiryTime: String("bar")}
+		opts.Set.SetForFlow.SetForBasicAuthentication = &SetForBasicAuthentication{Username: String("foo"), Password: String("bar")}
+		opts.Set.SetForFlow.SetForGenericString = &SetForGenericString{SecretString: String("secret")}
```
Comment (sfc-gh-jmichalak) [inline pkg/sdk/secrets_gen_test.go:205]: Nit: These 5 statements can be done like:
```
opts.Set.SetForFlow = &SetForFlow{
        SetForOAuthClientCredentials: &SetForOAuthClientCredentials{OauthScopes: &OauthScopesList{[]ApiIntegrationScope{{Scope: "foo"}}}}
        ...
}
```

**Review by sfc-gh-jmichalak — APPROVED** [review]

File: pkg/sdk/testint/secrets_gen_integration_test.go (hunk 1 above)
Comment (sfc-gh-fbudzynski) [reply pkg/sdk/testint/secrets_gen_integration_test.go:22]: It should, will fix this with next PR

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/sdk/testint/secrets_gen_integration_test.go (hunk 2 above)
Comment (sfc-gh-fbudzynski) [reply pkg/sdk/testint/secrets_gen_integration_test.go:123]: Will correct it with next Secret related PR

File: pkg/sdk/secrets_gen_test.go (hunk 3 above)
Comment (sfc-gh-fbudzynski) [reply pkg/sdk/secrets_gen_test.go:205]: will do this way next time and change it with next Secret PRs

**Pull request merged**